packages/waku/src/router/client.ts

```diff
@@ -125,16 +136,30 @@ export function useRouter_UNSTABLE() {
         '',
         url,
       );
-      changeRoute(parseRoute(url), { shouldScroll: newPath });
+      changeRoute(parseRoute(url), {
+        shouldScroll: options?.scroll ?? newPath,
+      });
     },
     [changeRoute],
   );
   const replace = useCallback(
-    (to: InferredPaths) => {
+    (
+      to: InferredPaths,
+      options?: {
+        /**
+         * indicates if the link should scroll or not on navigation
+         * - `true`: always scroll
+         * - `false`: never scroll
+         * - `undefined`: scroll on path change (not on searchParams change)
+         */
+        scroll?: boolean;
+      },
+    ) => {
       const url = new URL(to, window.location.href);
       const newPath = url.pathname !== window.location.pathname;
       window.history.replaceState(window.history.state, '', url);
-      changeRoute(parseRoute(url), { shouldScroll: newPath });
+      const shouldScroll = options?.scroll ?? newPath;
+      changeRoute(parseRoute(url), { shouldScroll });
```

Comment:
The coding style is not very consistent with `push` L140.

Reply:
```suggestion
      changeRoute(parseRoute(url), {
        shouldScroll: options?.scroll ?? newPath,
      });
```

Reply:
ah, good catch, fixed